__tests__/secret.test.js

```diff
@@ -0,0 +1,180 @@
+  username = `it-user-${uuidv4()}`
+  password = uuidv4()
+  realmConfig = {
+    realmName: idRealmName,
+    appName: idAppName,
+    brokerTargetUrl: 'http://integrationtest.local.tozny.com',
+    apiUrl,
+  }
+  realm = new Tozny.identity.Realm(
+    realmConfig.realmName,
+    realmConfig.appName,
+    realmConfig.brokerTargetUrl,
+    apiUrl
+  )
+  await realm.register(
+    username,
+    password,
+    clientRegistrationToken,
+    `${username}@example.com`
+  )
+  identity = await realm.login(username, password)
+})
+
+describe('Tozny identity client', () => {
+  it('can create a secret', async () => {
+    const secret = {
+      secretType: 'Credential',
+      secretName: `test-secret-${uuidv4()}`,
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    const secretTest = {
+      meta: {
+        type: `tozny.secret.${SECRET_UUID}.${secret.secretType}.${secret.secretName}`,
+        plain: {
+          description: secret.description,
+          secretName: secret.secretName,
+          secretType: secret.secretType,
+        },
+      },
+      data: {
+        secretValue: secret.secretValue,
+      },
+    }
+    const secretResp = await ops.createSecret(realmConfig, identity, secret)
+    expect(secretResp).toMatchObject(secretTest)
+  })
+  it('fails to create secret when name or value is not valid', async () => {
+    const secretTypeEmpty = {
+      secretType: '',
+      secretName: 'SecretName',
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    const secretTypeInvalid = {
+      secretType: 'Cred',
+      secretName: 'SecretName',
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    const secretNameEmpty = {
+      secretType: 'Credential',
+      secretName: '',
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    const secretNameInvalid = {
+      secretType: 'Credential',
+      secretName: `test-secret#-${uuidv4()}`,
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    const secretValueEmpty = {
+      secretType: 'Credential',
+      secretName: `test-secret-${uuidv4()}`,
+      secretValue: '',
+      description: 'this is a description',
+    }
+    expect(ops.createSecret(secretTypeEmpty)).rejects.toThrow()
+    expect(ops.createSecret(secretTypeInvalid)).rejects.toThrow()
+    expect(ops.createSecret(secretNameEmpty)).rejects.toThrow()
+    expect(ops.createSecret(secretNameInvalid)).rejects.toThrow()
+    expect(ops.createSecret(secretValueEmpty)).rejects.toThrow()
+  })
+  it('can create a secret, and list it', async () => {
+    const secret = {
+      secretType: 'Credential',
+      secretName: `test-secret-${uuidv4()}`,
+      secretValue: 'secret-value',
+      description: 'this is a description',
+    }
+    await ops.createSecret(realmConfig, identity, secret)
+    await new Promise(r => setTimeout(r, 1000))
```

Comment:
In an older PR I posted an example of a helper function which will keep these running faster and retry a number of times. I'd like to see us switch to something like that instead of just a 1 second timeout. I'm wondering if this may be part of the reason our tests are failing in some environments. 🤔 

ref: https://github.com/tozny/js-sdk/pull/41/files#r572278849

```js
/**
 * Get the results from a query, using a loop to ensure they have time to index
 */
async function waitForNext(query) {
  // Start with a very short delay as immediate fetch of results right after writing
  // sometimes failes, but even with a very short window of wait it can succeed
  // first try.
  await new Promise(r => setTimeout(r, i * 100))
  // 10-second timeout period, 20 rounds of half a second between tries.
  for (let i = 0; i < 20; i++) {
    const found = await query.next()
    if (found.length > 0) {
      break
    }
    // delay half a second between tries
    await new Promise(r => setTimeout(r, i * 500))
  }
  return found
}
```